src/Microsoft.DotNet.Build.CloudTestTasks/AssemblyResolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    /// <summary>
+    /// Used to enable custom assembly resolution.
+    /// </summary>
+    class AssemblyResolver
+    {
+        static int s_enabled;
+
+        /// <summary>
+        /// Call to enable the assembly resolver for the current AppDomain.
+        /// </summary>
+        public static void Enable()
+        {
+            if (Interlocked.CompareExchange(ref s_enabled, 1, 0) == 0)
```

Comment:
Avoid the flag entirely and just put the code in the static constructor for AssemblyResolver.  Have an Enable method that does nothing.  The act of calling it will trigger the static constructor which is guaranteed to be thread safe and run once.